dspace-api/src/main/java/org/dspace/content/service/ItemService.java

```diff
@@ -768,6 +769,27 @@ public Iterator<Item> findByLastModifiedSince(Context context, Date last)
      */
     int countWithdrawnItems(Context context) throws SQLException;
 
+    /**
+     * finds all items for which the current user has editing rights
+     * @param context DSpace context object
+     * @param offset page offset
+     * @param limit  page size limit
+     * @return list of items for which the current user has editing rights
+     * @throws SQLException
+     * @throws SearchServiceException
+     */
+    public List<Item> findItemsWithEdit(Context context, int offset, int limit)
```

Comment:
Could we add Tests for these new methods to the existing `org.dspace.content.ItemTest` class?  I just realized we're missing tests for `findItemsWithEdit()` and `countItemsWithEdit()`.

There are similar tests in that class for other `find*()` methods and even `canEdit()` for the ItemService. So, it'd be good to add a few small tests for these new methods.

Reply:
Hi @tdonohue, I added some tests to the `ItemServiceTest` class, as apparently there's no SOLR indexing available in the `ItemTest` tests. I kept the tests rather simple, as the full functionality is tested extensively in `EditItemFeatureIT`.